spec-tests/src/test/java/org/candlepin/spec/HypervisorCheckInSpecTest.java

```diff
@@ -763,14 +771,793 @@ public void shouldAllowASingleGuestToBeMigratedAndRevokeHostLimitedEntsHosted()
+        EntitlementDTO guestRH00051Entitlement = ApiClient.MAPPER.convertValue(ent, EntitlementDTO.class);
+        assertEquals(hypervisor1StackDerivedPool2, guestRH00051Entitlement.getPool().getId());
+        ent = guestClient.consumers().bindPool(guest.getUuid(), hypervisor1StackDerivedPool1, 1).get(0);
+        EntitlementDTO guestRH00271Entitlement = ApiClient.MAPPER.convertValue(ent, EntitlementDTO.class);
+        assertEquals(hypervisor1StackDerivedPool1, guestRH00271Entitlement.getPool().getId());
+
+        // Verify guest's entitlement certs each contain the appropriate content set
+        List<JsonNode> jsonNodes = guestClient.consumers().exportCertificates(guest.getUuid(), null);
+        assertEquals(2, jsonNodes.size());
+        JsonNode rh00051Cert = jsonNodes.stream().filter(x ->
+            guestRH00051Entitlement.getPool().getId().equals(x.get("pool").get("id").asText()))
+            .collect(Collectors.toList()).get(0);
+        JsonNode rh00271Cert = jsonNodes.stream().filter(x ->
+            guestRH00271Entitlement.getPool().getId().equals(x.get("pool").get("id").asText()))
+            .collect(Collectors.toList()).get(0);
+        assertNotNull(rh00051Cert);
+        assertNotNull(rh00271Cert);
+
+        assertEquals(1, rh00051Cert.get("products").get(0).get("content").size());
+        assertEquals(rh00051Content.getId(),
+            rh00051Cert.get("products").get(0).get("content").get(0).get("id").asText());
+        assertEquals(1, rh00271Cert.get("products").get(0).get("content").size());
+        assertEquals(rh00271Content.getId(),
+            rh00271Cert.get("products").get(0).get("content").get(0).get("id").asText());
+
+        executeMigration(owner, data1, data2, virtwho, guestUuid);
+        //  At this point, the guest will have the entitlements from the old hypervisor revoked, and
+        //  it will get the entitlements from the new hypervisor auto-attached
+        List<JsonNode> newJsonNodes = guestClient.consumers().exportCertificates(guest.getUuid(), null);
+        assertEquals(2, jsonNodes.size());
+        // Verify guest's entitlement certs each contain the appropriate content set
+        List<EntitlementDTO> newEnts = guestClient.consumers().listEntitlements(guest.getUuid());
+        assertEquals(2, newEnts.size());
+
+        EntitlementDTO updatedGuestRH00271Entitlement = newEnts.stream().filter(x ->
+            guestRH00271Entitlement.getPool().getProductId().equals(x.getPool().getProductId()))
+            .collect(Collectors.toList()).get(0);
+        assertNotNull(updatedGuestRH00271Entitlement);
+        EntitlementDTO updatedGuestRH00051Entitlement = newEnts.stream()
+            .filter(x -> guestRH00051Entitlement.getPool().getProductId().equals(x.getPool().getProductId()))
+            .collect(Collectors.toList()).get(0);
+        assertNotNull(updatedGuestRH00051Entitlement);
+
+        rh00051Cert = null;
+        rh00271Cert = null;
```

Comment:
You shouldn't need to assign to null since the are already being reassigned to the output of the stream filtering in the next couple of lines